ntp-proto/src/timetypes.rs

```diff
@@ -1,45 +1,69 @@
+use std::ops::Sub;
+
 #[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord)]
 pub struct NtpTimestamp {
     timestamp: u64,
 }
 
+impl Default for NtpTimestamp {
+    fn default() -> Self {
+        Self::from_bits([0u8; 8])
+    }
+}
+
 impl NtpTimestamp {
-    pub(crate) fn from_bits(bits: [u8; 8]) -> NtpTimestamp {
+    pub(crate) const fn from_bits(bits: [u8; 8]) -> NtpTimestamp {
         NtpTimestamp {
             timestamp: u64::from_be_bytes(bits),
         }
     }
 
-    pub(crate) fn to_bits(&self) -> [u8;8] {
+    pub(crate) const fn to_bits(self) -> [u8; 8] {
         self.timestamp.to_be_bytes()
     }
 
     #[cfg(test)]
-    pub(crate) fn from_fixed_int(timestamp: u64) -> NtpTimestamp {
+    pub(crate) const fn from_fixed_int(timestamp: u64) -> NtpTimestamp {
         NtpTimestamp { timestamp }
     }
 }
 
+impl Sub for NtpTimestamp {
+    type Output = NtpDuration;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        NtpDuration {
+            duration: self.timestamp as i64 - rhs.timestamp as i64,
+        }
+    }
+}
+
 #[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord)]
 pub struct NtpDuration {
     duration: i64,
 }
 
+impl Default for NtpDuration {
+    fn default() -> Self {
+        Self::from_bits_short([0u8; 4])
```

Comment:
This is somewhat convoluted, wouldn't a simple `NtpDuration { duration: 0 }` be better?